File: src/Application/Workflows/ReWoo/Dto/HotelDto.cs
namespace Application.Workflows.ReWoo.Dto;

public record HotelSearchResultDto
{
    public string ArtifactKey { get; init; }
    public List<HotelOptionDto> Results { get; init; }
}

public record HotelOptionDto
{
    public string HotelName { get; init; }
    public string Address { get; init; }
    public DateTime CheckIn { get; init; }
    public DateTime CheckOut { get; init; }
    public decimal Rating { get; init; }
    public HotelPriceDto PricePerNight { get; init; }
    public HotelPriceDto TotalPrice { get; init; }
}

public class HotelPriceDto
{
    public decimal Amount { get; set; }
    public string Currency { get; set; }
}

File: ui/src/types/dto/hotel.dto.ts
export interface HotelSearchResultDto {
    artifactKey: string;
    results: HotelOptionDto[];
}

export interface HotelOptionDto {
    hotelName: string;
    address: string;
    checkIn: string;
    checkOut: string;
    rating: number;
    pricePerNight: HotelPriceDto;
    totalPrice: HotelPriceDto;
}

export interface HotelPriceDto {
    amount: number;
    currency: string;
}